java/README.md

```diff
@@ -0,0 +1,101 @@
+# Java benchmarks
+
+## Verification Task Structure
+
+Verification tasks are grouped in directories, e.g.,
+`jbmc-regression`.  Within these directories, each verification task
+is in its own directory, e.g., `StringValueOf01_true-assert`. The
+suffix of this directory contains the expected verification outcome,
+i.e., `_true-assert` means that there is no execution of the program
+that violates any of the assertions in that program.  A verification
+task has a structure compatible with the [Maven](https://maven.apache.org)
+standard directory layout (to allow the future use of the Maven build
+system if necessary):
+
+| directory | description |
+| --------- | ----------- |
+|`src/` | |
+| `src/main/java/Main.java` | The `Main` class. |
+| `src/main/java/` | This directory contains the remaining Java source tree. |
+| `lib/` | This directory contains the library dependencies (without `java.*` and `javax.*`). |
+| `target/` | |
+| `target/classes/` | After compiling this directory contains the `.class` built from `src`. |
+| `target/task.jar` | This file assembles the `.class` files from `target/classes` and those from within the `.jar` files in `lib`. |
+
+The programs are assumed to be written in Java 1.8.
+The `.java` source files are in the source tree
+in the `src` sub-directory of the benchmark directory.  The
+program may call the Java standard library (`java.*`,
+`javax.*`).  Calls to other libraries are allowed under the
+condition that the library is provided as a `.jar` file in the
+`lib` sub-directory of the benchmark directory.  The benchmark
+must have a `Main` class with a `public static void
+  main(String[])` method in the root package. The `Main.java` file
+must have a copyright header indicating the source of the benchmark as
+well as name, version and links to the source code of all used libraries
+that are included in `lib`.
+
+## Compiling the Verification Tasks
+
+Makefiles for compiling the sources and assembling them into
+a `target/task.jar` file are provided through links to the
+`Makefile.task`.
+The top-level `Makefile` has a `compile` target for compiling all
+verification tasks. 
+
+## Preparing the Verification Tasks for Benchexec
+
+The input for a verification tool when run in Benchexec
+is a _verification-task-dir-name_`.zip` file, which contains
```

Comment:
> Where would these ZIPs come from?

`make prepare`

> Why not simply use the directory layout that is present in the repository?

You could use the Main.java file as "task" file that Benchexec uses to identify the benchmarks, but that's rather clumsy - so, I discarded that idea.

Reply:
Sorry, I can't follow. BenchExec does not need a file to identify tasks.

Reply:
I have tried. It cannot handle a verification task that is a directory. Maybe it is easy to fix, though.

Reply:
Sure it can, CPAchecker has been using this for years for its Java regression tests: https://github.com/sosy-lab/cpachecker/blob/trunk/test/test-sets/integration-explicitAnalysis-java.xml

But anyway, with the new metadata format there will be a `yml` file for each task, which will be BenchExec's way to identify a task, and that `yml` file will be able to refer to a directory of source files.

Reply:
> CPAchecker has been using this for years for its Java regression tests

Yes, but that requires listing all tasks in the benchmark-def, which is currently duplicated for each tool and therefore rather cumbersome. It would be better to manage this with the .set files within the sv-benchmarks repository.

Reply:
>  there will be a yml file for each task, which will be BenchExec's way to identify a task

I'm looking forward to this :)